Casks/puppet-bolt.rb

```diff
@@ -3,35 +3,63 @@
   when '10.11'
     os_ver = '10.11'
     version '2.0.0'
-    sha256 'a5956c7d075a6219f04cda8890732bb8e31b4e10ad0096e18046531b43cfdd7d'
+    if arch == 'arm64'
+      sha256 'nil'
+    elsif arch == 'x86_64'
+      sha256 'a5956c7d075a6219f04cda8890732bb8e31b4e10ad0096e18046531b43cfdd7d'
+    end
   when '10.12'
     os_ver = '10.12'
     version '2.0.0'
-    sha256 '9d14f5106bb1c882971658ca00537fd1f1e176685f31588637eab8b95feba0d7'
+    if arch == 'arm64'
+      sha256 'nil'
+    elsif arch == 'x86_64'
+      sha256 '9d14f5106bb1c882971658ca00537fd1f1e176685f31588637eab8b95feba0d7'
+    end
   when '10.13'
     os_ver = '10.13'
     version '2.0.0'
-    sha256 'ec644414592e24f685e41f696e9830958b5c2223b7d489038529520faf3d3352'
+    if arch == 'arm64'
+      sha256 'nil'
+    elsif arch == 'x86_64'
+      sha256 'ec644414592e24f685e41f696e9830958b5c2223b7d489038529520faf3d3352'
+    end
   when '10.14'
     os_ver = '10.14'
     version '3.13.0'
-    sha256 '22dc674608197d63c5c1bd199dc4337eff2822ea6cb9fefab3a1cc4ee3ec7187'
+    if arch == 'arm64'
+      sha256 'nil'
+    elsif arch == 'x86_64'
+      sha256 '22dc674608197d63c5c1bd199dc4337eff2822ea6cb9fefab3a1cc4ee3ec7187'
+    end
   when '10.15'
     os_ver = '10.15'
     version '3.22.1'
-    sha256 '4f17695d85a321260a5d60ab67a490c99cb27be9cdb83ef50e906be36163fb24'
+    if arch == 'arm64'
+      sha256 'nil'
+    elsif arch == 'x86_64'
+      sha256 '4f17695d85a321260a5d60ab67a490c99cb27be9cdb83ef50e906be36163fb24'
+    end
   when '11'
     os_ver = '11'
-    version '3.26.1'
-    sha256 '30d98cdc4aa9a5c28353d999b2968e625277756c87e1f49d83d0886b209d8064'
+    version '3.26.2'
+    if arch == 'arm64'
+      sha256 'nil'
+    elsif arch == 'x86_64'
+      sha256 'abb6482d9fa2c48e532339cd5fc0befe57d13f9e2530f4edc306657565e7d9c5'
+    end
   else
     os_ver = '12'
-    version '3.26.1'
-    sha256 '348f796687ac923651703e2e99a15e67fd2c9045b9b564e845501ab8f4f9720e'
+    version '3.26.2'
+    if arch == 'x86_64'
+      sha256 '3e86a1ce72ad1df62ba8491383f7c0a2c4513ed8359cd4973e9e27c98588f7e3'
+    elsif arch == 'arm64'
+      sha256 'nil'
+    end
   end
 
   depends_on macos: '>= :el_capitan'
-  url "https://downloads.puppet.com/mac/puppet-tools/#{os_ver}/x86_64/puppet-bolt-#{version}-1.osx#{os_ver}.dmg"
```

Comment:
I dont see where we define arch? I think we need some code like https://github.com/puppetlabs/homebrew-puppet/blob/7b9d7ef9ae2fc870e4589077045f16baec423c3b/Casks/puppet-agent-7.rb#L2-L6

Reply:
Oh, yes. I think I need to do this for all the non puppet-agent templates. Not certain about pe-client-tools.

Reply:
Fixed.